### PR TITLE
Dao and Services for Order, ShoppingCart, User were added

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: src/main/java/com/internet/shop/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User create(User user) {
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/UserServiceImpl.java]:
```suggestion
```

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        return shoppingCart.getProducts().remove(product);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java:46]: we should update shoppingCart after changes

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shoppingCart;
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        return shoppingCart.getProducts().remove(product);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: same her and in addProduct method

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
```

File: src/main/java/com/internet/shop/db/Storage.java [hunk 5]
```diff
@@ -1,15 +1,41 @@
...
+        shoppingCart.setId(shoppingCartId++);
+        shoppingCarts.add(shoppingCart);
+    }
+
+    public static void addOrder(Order order) {
+        order.setId(orderId++);
+        orders.add(order);
+    }
 }
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/db/Storage.java]:
```suggestion
```

**Review by RomanDubovskyi — COMMENTED** [review]
next time please push changes to the same PR after initial review, it's easier to review this way

File: src/main/java/com/internet/shop/db/Storage.java (hunk 5 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/db/Storage.java]: Done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: Done

File: src/main/java/com/internet/shop/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/service/impl/UserServiceImpl.java]: Done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 2 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java:46]: Done

File: src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java]: Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**